### PR TITLE
zigbee: Fix for scheduling zboss callbacks

### PR DESCRIPTION
**Pull request by mariuszpos**
Fix for scheduling zboss callbacks
Redefined ZBOSS macros (#214) to use thread safe functions.

This PR depends on: nrfconnect/sdk-nrfxlib#214

### CODE REVIEW

**Comment by lemrey** [comment]
It would be nice to settle on whether it's going to be `Zigbee` or `zigbee` when prefixing the commit messages :)

**Comment by wbober** [comment]
> It would be nice to settle on whether it's going to be `Zigbee` or `zigbee` when prefixing the commit messages :)

Plus the commit message should be a more descriptive, so that reviewer has some context what is being fixed.

File: subsys/zigbee/osif/zb_nrf_platform.c [hunk 1]
```diff
@@ -31,8 +35,10 @@ typedef enum {
 typedef struct {
 	zb_callback_type_t type;
 	zb_callback_t func;
-	zb_uint8_t param;
+	zb_callback2_t func2;
+	zb_uint16_t param;
 	zb_uint16_t user_param;
+	// zb_uint16_t user_param2;
```
Comment (tomchy) [inline subsys/zigbee/osif/zb_nrf_platform.c:41]: Leftover?

**Comment by mariuszpos** [comment]
> @mariuszpos @tomchy I'd prefer a solution that is more transparent to the user. There are several possibilities:
> a) introduce and redefine ZBOSS macros as in `ZB_SCHEDULE_APP_ALARM_CANCEL`
> b) make original ZBOSS functions weak
> c) use `--wrap` linker option to wrap around original ZBOSS functions

I've chosen the option: a) 
The code is updated.

**Comment by tomchy** [comment]
Do not forget to temporary update `west.yml` before running test suites :slightly_smiling_face:

**Comment by wbober** [comment]
@mariuszpos Where is the ```ZB_SCHEDULE_APP_ALARM``` is redefined?

**Comment by mariuszpos** [comment]
https://github.com/nrfconnect/sdk-nrfxlib/pull/206

**Comment by tomchy** [comment]
Please do the following fixes:
 - Rebase the branch as it is not building now.
 - Point to the updated ZBOSS library inside nrfxlib (https://github.com/nrfconnect/sdk-nrfxlib/pull/214)

**Comment by rlubos** [comment]
@mariuszpos Could you rebase please? We'd need a green CI on this one, and the conflict is preventing it from running properly. It seems that the CI issues we were facing yesterday are resolved now.

**Pull request merged**